app/src/main/java/com/github/se/travelpouch/ui/profile/ModifyingProfileScreen.kt

```diff
@@ -140,74 +148,163 @@ fun ModifyingProfileScreen(
+                                  label = { Text("Friend Email") },
+                                  placeholder = { Text("example@example.com") },
+                                  modifier = Modifier.testTag("addingFriendField"))
+
+                              Button(
+                                  onClick = {
+                                    if (friendMail == profile.value.email) {
+                                      Log.d("Friend added", "It is you")
+                                      Toast.makeText(
+                                              context, "You cannot add yourself", Toast.LENGTH_LONG)
+                                          .show()
+                                    } else if (profile.value.friends.contains(friendMail)) {
+                                      Log.d("Friend added", "Already friend with this user")
+                                      Toast.makeText(
+                                              context,
+                                              "You cannot add a friend you are already friend with",
+                                              Toast.LENGTH_LONG)
                                           .show()
-                                    },
-                                    onFailure = { e ->
-                                      Toast.makeText(context, e.message!!, Toast.LENGTH_LONG).show()
-                                    })
-                              }
-                            },
-                            modifier = Modifier.testTag("addingFriendButton")) {
-                              Text("Add Friend")
+                                    } else {
+                                      profileModelView.addFriend(
+                                          friendMail,
+                                          onSuccess = {
+                                            Toast.makeText(
+                                                    context, "Friend added", Toast.LENGTH_LONG)
+                                                .show()
+                                          },
+                                          onFailure = { e ->
+                                            Toast.makeText(context, e.message!!, Toast.LENGTH_LONG)
+                                                .show()
+                                          })
+                                    }
+                                  },
+                                  modifier = Modifier.testTag("addingFriendButton")) {
+                                    Text("Add Friend")
+                                  }
+                            }
+                      }
+                }
+              }
+
+              if (openDialogToDeleteFriend) {
+                Dialog(onDismissRequest = { openDialogToDeleteFriend = false }) {
+                  Box(
+                      Modifier.fillMaxWidth(1f)
+                          .height(250.dp)
```

Comment:
While it's the same size as for when adding a friend, the dialog box is too big for the content inside.
Either increase the size of the content or reduce the size of the dialog box. It feels weird.
![Screenshot_2024-12-05_20-52-01](https://github.com/user-attachments/assets/cacae485-98b0-4a14-9c8a-cda9c7e67eb1)

Reply:
I have corrected it